### PR TITLE
[bandit]: Do not treat "passed" as "password" for `S105`/`S106`/`S107`

### LINKED ISSUE #2384: Don't treat "passed" as "password" for `S105`/`S106`/`S107`
Using `v0.0.238` I get: `S105 Possible hardcoded password: "..."` when the variable/argument name contains `passed` in it.

It seems less likely that "passed" is being used for something related to passwords/secrets than not.

Example:
```python
passed_msg = "You have passed!"
failed_msg = "You have failed!"
```

Result:
```console
❯ ruff --select S1 bug.py 
bug.py:1:14: S105 Possible hardcoded password: "You have passed!"
Found 1 error.
```

Ahh... But if only you _had_ passed the test... 😏

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #2384

cc @KittyBorgX

### CODE REVIEW

File: crates/ruff/src/rules/flake8_bandit/helpers.rs [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 use rustpython_parser::ast::{Constant, Expr, ExprKind};
 
 use crate::checkers::ast::Checker;
 
-const PASSWORD_NAMES: [&str; 7] = [
-    "password", "pass", "passwd", "pwd", "secret", "token", "secrete",
-];
+static PASSWORD_CANDIDATE_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+    r"(^(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)$|_(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)_|^(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)_|_(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)$)"
```
Comment (not-my-profile) [inline crates/ruff/src/rules/flake8_bandit/helpers.rs:9]: There is absolutely no need to repeat `(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)` four times within this regex.

Comment (not-my-profile) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs:9]: I think the regex could be simplified to `\b(pas+wo?r?d|pass(phrase)?|pwd|token|secrete?)\b`.

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs:9]: I don't fully understand why they wrote it out in that way, but what @edgarrmondragon included here looks like a direct port of the version in Bandit: https://github.com/PyCQA/bandit/blob/a9eaafa53ae73c1a66172373128cb19bb7b82a32/bandit/plugins/general_hardcoded_password.py#L13-L16

Comment (edgarrmondragon) [reply crates/ruff/src/rules/flake8_bandit/helpers.rs]: You're right @charliermarsh, I just ported the Python regex.

@not-my-profile thanks for the suggestion, it makes total

**Pull request merged**